sagemathinc/cocalc
# webapp: fallback error message when react crashes

# Description
Hackish, but it works. 

![Screenshot from 2019-04-16 12-56-54](https://user-images.githubusercontent.com/207405/56204446-cc6ab300-6047-11e9-8d24-703588379820.png)

while inspecting the DOM, I found traces of "html" in `html.coffee` (duplicate content?) and templates for a console. I removed both and it still works. I'm not sure what it does, though, so I kept the console and removed the `html.coffee` file. 

# Testing Steps
1. I trigger this by adding a problematic line in `frame-tree/title-bar.tsx`: `  render_format(): Rendered  { return x.y; ... }` and obviously ignoring the TS error. When I open a file a crash happens. But pick any poison you like …
2. clicking on the "refresh" link refreshes and the help@ email address should be correct.
3. while loading the app, neither in desktop nor in mobile that box shows up.


# Relevant Issues

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
